### PR TITLE
Task/apps 2655 fix chronicle server downloads

### PR DESCRIPTION
**Pull request by anzioka**
implement Iterable interface

### CODE REVIEW

**Review by makosblade — CHANGES_REQUESTED** [review]
I like the general structure of this, using a supplier, iterator and iterable but it should definitely be written in Kotlin. There's no reason that we should be adding files written in Java anywhere in the codebase.

File: src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.java [hunk 1]
```diff
@@ -0,0 +1,257 @@
+package com.openlattice.chronicle.services.download;
```
[the remark this answers is not in the document]
Comment (anzioka) [reply src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.java]: done!

File: src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+                            .getOrDefault(EdmConstants.TIMEZONE_FQN, ImmutableSet.of<Any>(OutputConstants.DEFAULT_TIMEZONE))
+                            .iterator()
+                            .next()
+                            .toString()
+                    )
+                    entities.forEach { (key: FullQualifiedName?, `val`: Set<Any?>) ->
+                        if (!neighborPageSupplier.srcPropertiesToExclude.contains(key)) {
+                            val propertyType = neighborPageSupplier.edmCacheManager
+                                    .getPropertyType(key)
+                            val title = OutputConstants.APP_PREFIX + neighborPageSupplier.srcMetadata[propertyType.id]
```
Comment (geekbeast) [inline src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt:140]: title is nullable here. If it shouldn't be nullable use .getValue(...) instead [] index operator which will thrown an NPE if the property type id is not present.

File: src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+                    // 2: process association data
+                    val associations = neighborEntityDetails[currentIndex]
+                            .associationDetails
+                    associations.forEach { (key: FullQualifiedName?, `val`: Set<Any?>) ->
+                        if (!neighborPageSupplier.edgePropertiesToExclude.contains(key)) {
+                            val propertyTypeId = neighborPageSupplier.edmCacheManager
+                                    .getPropertyTypeId(key)
+                            val title = OutputConstants.USER_PREFIX + neighborPageSupplier.edgeMetadata[propertyTypeId]
+                                    ?.title
```
Comment (geekbeast) [inline src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt:158]: ditto.

File: src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                    result.add(parsed)
+                } catch (e: Exception) {
+                    result.add("")
+                }
+            })
+            return result
+        }
+
+        override fun next(): Map<String, Set<Any>> {
+            lock.lock()
```
Comment (geekbeast) [inline src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt:105]: I don't understand the purpose of this lock. It isn't used outside of this function and seems to serve a synchronization mechanism for the method. Use ```@Synchronized``` annotation instead.

**Review by geekbeast — CHANGES_REQUESTED** [review]
Looks good, a requested change (synchronization) and some minor suggestions.

**Review by makosblade — APPROVED** [review]

**Review by katherinebernstein — APPROVED** [review]
This is awesome!! I left a few tiny comments on how to kotlinify some things, and one bigger suggestion about structuring things in a more readable way, but this looks great and none of those things should be blocking for getting this fix in! (Mostly intended as suggestions for updates/refactors for the future)

**Review by geekbeast — APPROVED** [review]

**Pull request merged**